iwyu.cc

```diff
@@ -1657,23 +1657,14 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     type = MapPrivateTypeToPublicType(type);
     // For the below, we want to be careful to call *our*
     // ReportDeclUse(), not any of the ones in subclasses.
-    if (IsPointerOrReferenceAsWritten(type)) {
```

Comment:
Maybe, leave here
```c++
if (IsPointerOrReferenceAsWritten(type))
  return;
```
just for clarity and robustness? Or redundant code should be avoided? @kimgr, what do you think?

Reply:
My preference is for an assertion instead, so we can learn when the assumption is broken: `CHECK_(!IsPointerOrReferenceAsWritten(type));  // this should already be handled in ...`

Reply:
Or wait, there is no separate handling?

Reply:
Are you saying we never get here for pointer types, or that the casting below somehow ignores them?

Reply:
Pointers may get here, as in the cases of indexing of array of pointers, or taking `sizeof` variable of pointer type, but they should be ignored.

Reply:
Yes, `TypeToDeclAsWritten` returns `nullptr` for them, at least at now.

Reply:
Yeah, I think it makes sense to short-circuit for pointer/references then, with the if-check you suggested.

But it would be nice to spell out in a comment why it's unnecessary, and who will cover the reporting instead (if anyone). It wouldn't surprise me if there's some construct we don't have test coverage for that will be missed now that this is removed.